File: backend/domains/user/http/router.go
package http

import (
	"github.com/cloudwego/hertz/pkg/route"
	"github.com/erweixin/go-genai-stack/backend/domains/user/handlers"
	"github.com/erweixin/go-genai-stack/backend/infrastructure/middleware"
)

// RegisterRoutes 注册 User 领域的路由
//
// 参数：
//   - r: 路由分组
//   - deps: Handler 依赖容器
//   - authMiddleware: 认证中间件
//
// 路由列表：
//   - GET    /users/me                   - 获取当前用户资料（需要认证）
//   - PUT    /users/me                   - 更新当前用户资料（需要认证）
//   - POST   /users/me/change-password   - 修改密码（需要认证）
func RegisterRoutes(
	r *route.RouterGroup,
	deps *handlers.HandlerDependencies,
	authMiddleware *middleware.AuthMiddleware,
) {
	// User API 分组（需要认证）
	userGroup := r.Group("/users")
	userGroup.Use(authMiddleware.Handle())
	{
		// 获取当前用户资料
		userGroup.GET("/me", deps.GetUserProfileHandler)

		// 更新当前用户资料
		userGroup.PUT("/me", deps.UpdateUserProfileHandler)

		// 修改密码
		userGroup.POST("/me/change-password", deps.ChangePasswordHandler)
	}
}

File: backend-nodejs/domains/user/http/router.ts
/**
 * User 路由注册
 * 注册所有 User 相关的 HTTP 路由
 */

import type { FastifyInstance } from 'fastify';
import type { HandlerDependencies } from '../handlers/dependencies.js';
import type { UpdateUserProfileRequest, ChangePasswordRequest } from './dto/user.js';
import { UpdateUserProfileRequestSchema, ChangePasswordRequestSchema } from './dto/user.js';
import { getUserProfileHandler } from '../handlers/get_user_profile.handler.js';
import { updateUserProfileHandler } from '../handlers/update_user_profile.handler.js';
import { changePasswordHandler } from '../handlers/change_password.handler.js';
import type { createAuthMiddleware } from '../../../infrastructure/middleware/auth.js';

/**
 * 注册 User 路由
 */
export function registerUserRoutes(
  app: FastifyInstance,
  deps: HandlerDependencies,
  authMiddleware: ReturnType<typeof createAuthMiddleware>
): void {
  // GET /api/users/me - 获取当前用户资料（需要认证）
  app.get('/api/users/me', { preHandler: authMiddleware }, async (req, reply) => {
    await getUserProfileHandler(deps, req, reply);
  });

  // PUT /api/users/me - 更新用户资料（需要认证）
  app.put<{ Body: UpdateUserProfileRequest }>(
    '/api/users/me',
    {
      preHandler: authMiddleware,
      schema: {
        body: UpdateUserProfileRequestSchema,
      },
    },
    async (req, reply) => {
      await updateUserProfileHandler(deps, req, reply);
    }
  );

  // POST /api/users/me/change-password - 修改密码（需要认证）
  app.post<{ Body: ChangePasswordRequest }>(
    '/api/users/me/change-password',
    {
      preHandler: authMiddleware,
      schema: {
        body: ChangePasswordRequestSchema,
      },
    },
    async (req, reply) => {
      await changePasswordHandler(deps, req, reply);
    }
  );
}
